### PR TITLE
Add spotless plugin to check google-java-format compliance

### PR DESCRIPTION
**Pull request by mprashanthsagar**

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -99,6 +99,7 @@
     <google.http-client.version>1.39.2-sp.1</google.http-client.version>
     <google.oauth-client.version>1.31.5</google.oauth-client.version>
     <google.protobuf.version>3.17.3</google.protobuf.version>
+    <spotless.version>2.12.2</spotless.version>
```
Comment (medb) [inline pom.xml:102]: Usually we specify plugin versions in-line.

File: pom.xml [hunk 2]
```diff
@@ -699,6 +705,41 @@
...
         </configuration>
       </plugin>
+      <plugin>
+        <groupId>com.diffplug.spotless</groupId>
+        <artifactId>spotless-maven-plugin</artifactId>
+        <version>${spotless.version}</version>
+        <configuration>
+          <ratchetFrom>master</ratchetFrom>
+          <formats>
+            <!-- you can define as many formats as you want, each is independent -->
```
Comment (medb) [inline pom.xml:715]: Remove comments?

File: pom.xml [hunk 3]
```diff
@@ -699,6 +705,41 @@
...
+              </includes>
+              <!-- define the steps to apply to those files -->
+              <trimTrailingWhitespace/>
+              <endWithNewline/>
+            </format>
+          </formats>
+          <java>
+            <googleJavaFormat>
+              <version>1.7</version>
+              <style>AOSP</style>
```
Comment (medb) [inline pom.xml:730]: Seems like we need to remove this option, it's for Android: https://github.com/google/google-java-format/wiki/FAQ#i-just-need-to-configure-it-a-bit-differently-how

**Review by medb — CHANGES_REQUESTED** [review]

File: pom.xml [hunk 4]
```diff
@@ -481,6 +482,11 @@
         <version>${google.flogger.version}</version>
         <scope>runtime</scope>
       </dependency>
+      <dependency>
+        <groupId>com.diffplug.spotless</groupId>
+        <artifactId>spotless-maven-plugin</artifactId>
+        <version>${spotless.version}</version>
+      </dependency>
```
[the remark this answers is not in the document]
Comment (mprashanthsagar) [reply pom.xml:489]: Yeah, the [docs](https://github.com/diffplug/spotless/tree/main/plugin-gradle#google-java-format) specify it

File: pom.xml (hunk 1 above)
Comment (mprashanthsagar) [reply pom.xml:102]: Ack

File: pom.xml [hunk 5]
```diff
@@ -699,6 +705,41 @@
...
+                <include>.gitignore</include>
+              </includes>
+              <!-- define the steps to apply to those files -->
+              <trimTrailingWhitespace/>
+              <endWithNewline/>
+            </format>
+          </formats>
+          <java>
+            <googleJavaFormat>
+              <version>1.7</version>
```
[the remark this answers is not in the document]
Comment (mprashanthsagar) [reply pom.xml:729]: Post 1.7, googleJavaFormat has a dependency on Java 11

File: pom.xml (hunk 3 above)
Comment (mprashanthsagar) [reply pom.xml:730]: Good catch!

File: pom.xml (hunk 4 above)
[the remark this answers is not in the document]
Comment (medb) [reply pom.xml:488]: Does not look like Maven Plugin doc has it: https://github.com/diffplug/spotless/tree/main/plugin-maven#quickstart

Will it fail without this explicit dependency?

**Review by medb — APPROVED** [review]

**Pull request merged**